### ISSUE TITLE
Client randomly timeout and won't recover

### ISSUE THREAD

**Issue by nicreve** [redisson/redisson#2586]
Hi, we have some redisson client timeout issue.
We run a service with 30+ JVM instances. some instances' redisson client will randomly timeout (about 2 - 5 instances) and won't recover, but other instances still works fine.  The failed instances also use Jedis for some legacy feature, and it works fine too. We had to reboot the failed instances so it can recover. 
Redis servers are in cluster mode, we recently scaled our cluster from 3 masters, 3 slaves to 9 masters, 9 slaves, and the problem happens more often.
Here's the timeout exception, hope it helps:
```java
org.redisson.client.RedisTimeoutException: Unable to send command!
Node source: NodeSource [slot=null, addr=null, redisClient=null, redirect=null, 
entry=MasterSlaveEntry [masterEntry=[freeSubscribeConnectionsAmount=1, freeSubscribeConnectionsCounter=50, freeConnectionsAmount=40, freeConnectionsCounter=64, freezed=false, freezeReason=null, 
client=[addr=redis://10.30.123.42:26457], nodeType=MASTER, firstFail=0]]], 
connection: RedisConnection@742197929 [redisClient=[addr=redis://10.30.93.48:26457], 
channel=[id: 0xb70c2d91, L:/10.30.67.79:45357 - R:10.30.93.48/10.30.93.48:26457]], 
command: (GET), 
command params: [CATEGORY::OUTPUT::3_1] after 3 retry attempts
```
Redis server version: 3.0.7
Redisson version: 3.10.0

**Comment by mrniko** [comment]
Please refer to this section
https://github.com/redisson/redisson/wiki/16.-FAQ#q-i-saw-a-redistimeoutexception-what-does-it-mean-what-shall-i-do-can-redisson-team-fix-it

**Comment by mrniko** [comment]
In your case increasing of netty threads setting should help

**Comment by nicreve** [comment]
> In your case increasing of netty threads setting should help

@mrniko Hi niko, sorry but I don't think that's the case.
I analyzed the exception logs for one node and they all related to the same redis server node.
Maybe the connection and/or the corresponding netty encode/decode thread are blocked in someway?

**Comment by mrniko** [comment]
> I analyzed the exception logs for one instance and they all related to the same redis server node.

I think connection might too slow.

**Comment by semistone** [comment]
maybe similar to 
https://github.com/redisson/redisson/issues/2587

**Comment by mrniko** [comment]
Fixed in #2587
